Improved error logging for win_task.create_task_from_xml

Lesvek opened the pull request:
### What does this PR do?
Instead of only returning True or False when create_task_from_xml fail to add the scheduled task, the actual error message is returned to the user. If the error is unknown, the hex error code is returned. That way, it's easier to search for as this is the format Microsoft is using in their documentation.

### Previous Behavior
Previously, when there was an error, the module only returned False, only logging the error message in the debug logs.

### New Behavior
A proper exception containing the error message is now returned to the user when there is an error. New code will have to catch the exception.

### Tests written?
I added 2 new tests.

### Commits signed with GPG?
The commit was not signe with GPG.

Lesvek committed:
Improved error handling in win_task.create_task_from_xml
win_task.create_task_from_xml now returns the error message when it fails to add the scheduled task to make it easier to diagnose the problem.

Lesvek committed:
Merge remote-tracking branch 'origin/develop' into develop

Lesvek committed:
Changed the documentation to reflect the change. win_task.create_task_from_xml can now return a string when there was an error
Fixed a bug where win_task.create_task_from_xml did not worked when the username was set to None

Lesvek committed:
Added an integration test for win_task.create_task_from_xml

twangboy commented on salt/modules/win_task.py:
```diff
@@ -621,21 +622,38 @@ def create_task_from_xml(name,
+            failure_code = 'Unknown Failure: {0}'.format(error_code)
+        finally:
+            log.debug('Failed to create task: %s', failure_code)
+        return failure_code
```
We should probably raise a CommandExecutionError here instead. If the Return type is Bool, it should always be Bool. It shouldn't return 2 different types.

Lesvek replied:
Oh, nice. I didn't know it existed.

twangboy commented on salt/modules/win_task.py:
```diff
@@ -566,8 +566,8 @@ def create_task_from_xml(name,
-    :return: True if successful, False if unsuccessful
-    :rtype: bool
+    :return: True if successful, False if unsuccessful, A string with the error message if there is an error
+    :rtype: bool or str
```
Add a `:raises: CommandExecutionError` for completenes

Lesvek committed:
Raise a CommandExecutionError instead of returning the error message, as suggested in the core review.

Lesvek committed:
Corrected the win_task.create_task_from_xml test to fit with the new API.

twangboy commented on tests/integration/modules/test_win_task.py:
```diff
@@ -0,0 +1,76 @@
+from tests.support.unit import skipIf
+from tests.support.helpers import destructiveTest
+
+import salt.utils.platform
```
And you need two blank lines between the imports and the class (lint)

twangboy commented on tests/integration/modules/test_win_task.py:
```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
```
You probably need to add this to fix a lint error:
```
from __future__ import absolute_import, unicode_literals
```

twangboy commented on tests/integration/modules/test_win_task.py:
```diff
@@ -0,0 +1,76 @@
+        Test adding a task using xml
+        '''
+        xml_text = r"""<Malformed"""
+        with self.assertRaises(salt.exceptions.CommandExecutionError):
```
```
with self.assertRaises(CommandExecutionError):
```

Lesvek committed:
Use `from ... import` for the salt exceptions, as recommended
Fixed the integration test coding style error
Directly use the task module instead of using run_function. We didn't need the extra feature it provided

twangboy commented:
@Lesvek Thanks for getting all those things fixed and thanks for the contribution! Looks good now.
